src/Validation/Constraints/LogicalOr.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Validation\Constraints;
+
+use ILIAS\Validation\Constraint;
+use ILIAS\Data;
+
+/**
+ * Class LogicalOr
+ * @package ILIAS\Validation\Constraints
+ * @author  Michael Jansen <mjansen@databay.de>
+ */
+class LogicalOr extends Custom implements Constraint
+{
+	/**
+	 * @var Constraint[]
+	 */
+	protected $other = [];
+
+	/**
+	 * @var Constraint[]
+	 */
+	protected $failedOthers = [];
+
+	/**
+	 * LogicalOr constructor.
+	 * @param Constraint[] $other
+	 * @param Data\Factory $data_factory
+	 */
+	public function __construct(array $other, Data\Factory $data_factory)
+	{
+		$this->other = $other;
+
+		parent::__construct(
+			function ($value) {
+				$return = false;
+
+				foreach ($this->other as $constraint) {
+					if ($constraint->accepts($value)) {
+						$return = true;
+					} else {
+						$this->failedOthers[] = $constraint;
+					}
+				}
+
+				return $return;
+			},
+			function ($value) {
+				$problems = [];
+
+				foreach ($this->failedOthers as $constraint) {
+					$problems[] = (string)$constraint->problemWith($value);
+				}
+
+				return implode('', array_filter($problems));
+			},
```

Comment:
I see the problem... One solution could maybe be to get rid of primitive **problem** descriptions and move to some kind of problem abstraction with different types:
* OneOf (could be used by the constraint suggested with this PR)
* All (could be e.g. used by `\ILIAS\Validation\Constraints\Parallel`)
* Single (could be used by every other/simple constraint)

This could be interpreted by some kind of `ProblemRenderer`.

Reply:
Maybe: "Please fix one of these: ..." / "Bitte behebe eines der folgenden Probleme: ..."